src/consensus/params.h

```diff
@@ -25,6 +37,10 @@ struct Params {
     int64_t nPowTargetSpacing;
     int64_t nPowTargetTimespan;
     int64_t DifficultyAdjustmentInterval() const { return nPowTargetTimespan / nPowTargetSpacing; }
+    uint64_t MaxBlockSize(uint64_t nCurrentTime) const { return blockParams->MaxBlockSize(nCurrentTime); }
```

Comment:
But that means that either:
- The function will be the same for all chains
- MaxBlockSize will have to do a switch/case on the params and behave appropriately

The first way is out, since it means that (for ex) testnet and mainnet must share behavior for parameters
The second way gets ugly as additional chains are added.

I added them as virtuals so that each chain can specify its own behavior.